src/libcommon/log/sentry/sentryhandler.h

```diff
@@ -0,0 +1,105 @@
+/*
+ * Infomaniak kDrive - Desktop
+ * Copyright (C) 2023-2024 Infomaniak Network SA
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include <sentry.h>
+#include <string>
+#include <mutex>
+
+#include "utility/types.h"
+#include "sentryuser.h"
+
+namespace KDC {
+
+enum class SentryLevel {  // Not defined in types.h as we don't want to include sentry.h in types.h
+    Debug = SENTRY_LEVEL_DEBUG,
+    Info = SENTRY_LEVEL_INFO,
+    Warning = SENTRY_LEVEL_WARNING,
+    Error = SENTRY_LEVEL_ERROR,
+    Fatal = SENTRY_LEVEL_FATAL
+};
+
+inline std::string toString(SentryLevel level) {
+    switch (level) {
+        case SentryLevel::Debug:
+            return "Debug";
+        case SentryLevel::Info:
+            return "Info";
+        case SentryLevel::Warning:
+            return "Warning";
+        case SentryLevel::Error:
+            return "Error";
+        case SentryLevel::Fatal:
+            return "Fatal";
+    }
+};
+
+class SentryHandler {
+    public:
+        enum class SentryProject { Server, Client, Disable };  // Only used for initialization, don't need to be in types.h
```

Comment:
What about `Disabled` or `Deactivated` instead?

Reply:
Done here, I renamed it to Desactivated: [84f9bf5](https://github.com/Infomaniak/desktop-kDrive/pull/274/commits/84f9bf50bc1055ca21758988ebbc7fd0165c8b53)

Reply:
Your certainly meant `Deactivated`, see https://www.wordreference.com/enfr/deactivated.